### ISSUE TITLE
X11 Clipboard incompatible with KDE's default "Prevent empty clipboard"

### ISSUE THREAD

**Issue by falkTX** [lv2/pugl#72]
`puglSetClipboard` does not work on stock KDE, because of the clipboard tool it uses to ensure if one closes an application the clipboard data is not lost.

Specifically the "Prevent empty clipboard" breaks pugl

![Screenshot_20211209_224143](https://user-images.githubusercontent.com/1334853/145487308-0a225a31-b19b-4d47-ae44-68a8046217dc.png)

Trying to find out more...

**Comment by falkTX** [comment]
Actually now getting a soft-lock around

```
    // Run event loop until data is received
    while (!view->clipboard.data) {
      puglUpdate(view->world, -1.0);
    }
```

Current x11 clipboard handling code seems not very robust. The macOS part works fine. Didn't test in windows

**Comment by falkTX** [comment]
actually even after disabling the other KDE options I can't get this to work still.
pasting from other programs into pugl works fine, copying out of pugl does not.

**Comment by drobilla** [comment]
Weird.  I didn't try with anything that does snoopy tricks with the clipboard.  I'll try it with KDE.

**Comment by drobilla** [comment]
Seems to mostly work for me, with pugl_clipboard_demo and Konsole as a target, although the current code on `master` with that busy loop you pointed out is obviously problematic anyway.  Try the `dragdrop` branch (there's a pugl_clipboard_demo there), it reimplements all the X clipboard stuff (hopefully) correctly without any weird synchronous hacks.  The clipboard API is different, though.

**Comment by falkTX** [comment]
You tested on the `dragdrop` branch and not with master, correct?

**Comment by drobilla** [comment]
I tested both.

**Comment by drobilla** [comment]
I tore out the basic clipboard stuff from the dragdrop branch.  This should be fixed in 25012188, along with presumably many other X11 clipboard issues.  It's trickier to receive data now, but that's just the way it needs to work for type negotiation.
